project_task_description_template/models/project_task.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+from odoo import api, fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    task_template_id = fields.Many2one(
+        comodel_name="project.task.template",
+        compute="_compute_template_visible",
+        store=True,
+        readonly=False,
+    )
+    template_visible = fields.Boolean(compute="_compute_template_visible", store=True)
+
+    @api.depends("project_id", "stage_id")
+    def _compute_template_visible(self):
+        template_task_type_ids = self.mapped("project_id.template_task_type_ids")
+        for rec in self:
+            visible = not rec.project_id.is_restrict_template_by_stages or (
+                rec.stage_id
+                and rec.project_id
+                and template_task_type_ids
+                and rec.stage_id.id in template_task_type_ids.ids
+            )
+            rec.template_visible = visible
+            if visible and rec.project_id.default_task_template_id:
+                rec.task_template_id = rec.project_id.default_task_template_id
+                self._onchange_task_template_id()
```

Comment:
**issue**: I think there's a risk here that, when you change the stage of a task, the template is also changed automatically, and data is overwritten.

I'm thinking:
1. Project has "Stage 1" and "Stage 2" for its tasks 
2. Template is only active for stage 2
3. Create a task in stage 1, fill the details
4. Advance to stage 2

Here, the template would be automatically set, and fields would be overwritten.

Reply:
Thanks for review
I fixed this bug
check please)

Reply:
I checked on runboat, and it seems ok.